Sources/Commands/Describe.swift

```diff
@@ -10,85 +10,354 @@
+            self.targets = product.targets.map { $0.name }
+        }
     }
 
-    public func toJSON() -> JSON {
-        return .init([
-            "name": name,
-            "c99name": c99name,
-            "type": type,
-            "module_type": String(describing: Swift.type(of: self)),
-            "path": sources.root,
-            "sources": sources
-        ])
+    /// Represents a target for the sole purpose of generating a description.
+    struct DescribedTarget: Encodable {
+        let name: String
+        let type: String
+        let c99name: String?
+        let moduleType: String?
+        let path: String
+        let sources: [String]
+        let resources: [PackageModel.Resource]?
+        
+        init(from target: Target, in package: Package) {
+            self.name = target.name
+            self.type = target.type.rawValue
+            self.c99name = target.c99name
+            self.moduleType = String(describing: Swift.type(of: target))
+            self.path = target.sources.root.relative(to: package.path).pathString
+            self.sources = target.sources.relativePaths.map{ $0.pathString }
+            self.resources = target.resources.isEmpty ? nil : target.resources
+        }
     }
 }
 
-extension Sources: JSONSerializable {
-    public func toJSON() -> JSON {
-        return .array(relativePaths.map({ .string($0.pathString) }))
+
+public struct PlainTextEncoder {
+    
+    /// The formatting of the output plain-text data.
+    public struct FormattingOptions: OptionSet {
+        public let rawValue: UInt
+
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+
+        /// Produce plain-text format with indented output.
+        public static let prettyPrinted = FormattingOptions(rawValue: 1 << 0)
     }
-}
 
-extension Target.Kind: JSONSerializable {
-    public func toJSON() -> JSON {
-        return .string(rawValue)
+    /// The output format to produce. Defaults to `[]`.
+    var formattingOptions: FormattingOptions = []
+
+    /// Contextual user-provided information for use during encoding.
+    var userInfo: [CodingUserInfoKey: Any] = [:]
+
+    /// Encodes the given top-level value and returns its plain text representation.
+    ///
+    /// - parameter value: The value to encode.
+    /// - returns: A new `Data` value containing the encoded plan-text data.
+    /// - throws: An error if any value throws an error during encoding.
+    func encode<T: Encodable>(_ value: T) throws -> Data {
+        let outputStream = BufferedOutputByteStream()
+        let encoder = _PlainTextEncoder(outputStream: outputStream, formattingOptions: formattingOptions, userInfo: userInfo)
+        try value.encode(to: encoder)
+        return Data(outputStream.bytes.contents)
+    }
+    
+    /// Private helper function to format key names with an uppercase initial letter and space-separated components.
+    private static func displayName(for key: CodingKey) -> String {
```

Comment:
There isn't a lot of customizability here over the formatting.  In particular, it converts things like "URL" to "Url" which might be undesirable.